src/modules/Builder/features/ActivityItemsFlow/ItemFlow/SummaryRow/SummaryRow.tsx

```diff
@@ -12,10 +13,21 @@ import { getMatchOptions, getItemsOptions } from './SummaryRow.utils';
 
 export const SummaryRow = ({ name }: SummaryRowProps) => {
   const { t } = useTranslation('app');
-  const { control, watch } = useFormContext();
+  const { control, watch, setValue } = useFormContext();
 
   const { fieldName } = useCurrentActivity();
   const items = watch(`${fieldName}.items`);
+  const selectedItem = watch(`${name}.itemKey`);
+
+  useEffect(() => {
+    if (selectedItem) {
```

Comment:
pls use early return
```suggestion
    if (!selectedItem) return;
```

Reply:
done